extensions/notification/utils/config.js

```diff
@@ -0,0 +1,89 @@
+const constants = require('../constants');
+
+/**
+ * Transforms array of configurations to a Map
+ * @param  {Object[]} configs - Array of bucket notification configurations
+ * @return {Map} Map of bucket notification configurations
+ */
+function transform(configs) {
+    const configMap = new Map();
+    configs.forEach(config => {
+        if (config.bucket && config.notificationConfiguration) {
+            const bucket = config.bucket;
+            const conf = config.notificationConfiguration.queueConfigurations;
+            if (configMap.has(bucket)) {
+                const val = configMap.get(bucket);
+                configMap.set(bucket, [...val, ...conf]);
+            } else {
+                configMap.set(bucket, [...conf]);
+            }
+        }
+    });
+    return configMap;
+}
+
+/**
+ * Validate object name against bucket notification configuration filters
+ * @param  {Object[]} filterRules - Filter rules from the bucket notification
+ * configuration.
+ * @param  {string} objectName - Name of the object
+ * @return {boolean} true if filter matches the object name
+ */
+function validateFilter(filterRules, objectName) {
+    return filterRules.some(rule => {
+        const name = rule.name;
+        const value = rule.value;
+        if (name === constants.nameFilter.prefix) {
+            return objectName.startsWith(value);
+        }
+        if (name === constants.nameFilter.suffix) {
+            return objectName.endsWith(value);
+        }
+        return false;
+    });
+}
+
+/**
+ * Validate event against bucket notification configuration event type
+ * @param  {Object[]} bnConfig - Array of bucket specific notification
+ * configurations
+ * @param  {string} event - Type of event
+ * @return {undefined|Object} A bucket notification configuration if found
+ */
+function validateEvent(bnConfig, event) {
+    const config = bnConfig.filter(config => config.events.includes(event));
+    return config.length ? config[0] : undefined;
+}
+
+/**
+ * Validates an entry from log against bucket notification configurations to see
+ * if the entry has to be published. Validations include, bucket specific
+ * configuration check, event type check, object name specific filter checks
+ * @param  {Object[]} bnConfigs - Array of bucket notifications configuration
+ * @param  {string} bucket - Bucket name
+ * @param  {string} key - Object name
+ * @param  {string} event - Event type
+ * @return {boolean} true if event qualifies for notification
+ */
+function validateEntry(bnConfigs, bucket, key, event) {
+    // check if the entry belongs to any bucket in the configuration!
+    if (bnConfigs.has(bucket)) {
+        const bnConfig = bnConfigs.get(bucket);
+        // check if the event type matches!
+        const qConfig = validateEvent(bnConfig, event);
+        if (qConfig) {
```

Comment:
to reduce the indentation level a bit you could reverse with `if (!qConfig) { return false }; ...`

Reply:
I kept this simple, it was easier to read.

Reply:
OK, up to you it's a minor point